lib/certgen/certgen.go

```diff
@@ -136,13 +138,39 @@ func publicKey(priv interface{}) interface{} {
 	switch k := priv.(type) {
 	case *rsa.PrivateKey:
 		return &k.PublicKey
-	//case *ecdsa.PrivateKey:
+	// TODO: eventaully we need to suport ecdsa for CA
+	// case *ecdsa.PrivateKey:
 	//	return &k.PublicKey
 	default:
 		return nil
 	}
 }
 
+func ValidatePublicKeyStrength(pub interface{}) (bool, error) {
+	switch k := pub.(type) {
+	case *rsa.PublicKey:
+
+		if k.Size() < 256 { //ksize is in bytes
+			return false, nil
+		}
+
+		if k.E < 65537 {
+			return false, nil
+		}
+		return true, nil
+	case *ecdsa.PublicKey:
+		// TODO: check for the actual curves used
```

Comment:
why is that a todo? i think 256 and above is all ok, no?

Reply:
Not all curves are created equal, some have some bad properties (had to constant time or other algebraic badness). 256 bit size is necesary but not sufficient to have a secure curve (as of 2019)